api/agent.go

```diff
@@ -1316,10 +1324,21 @@ func (a *Agent) updateToken(target, token string, q *WriteOptions) (*WriteMeta,
 	return meta, err
 }
 
-func (a *Agent) updateTokenFallback(target, fallback, token string, q *WriteOptions) (*WriteMeta, error) {
-	meta, status, err := a.updateTokenOnce(target, token, q)
-	if err != nil && status == 404 {
-		meta, _, err = a.updateTokenOnce(fallback, token, q)
+func (a *Agent) updateTokenFallback(token string, q *WriteOptions, targets ...string) (*WriteMeta, error) {
+	if len(targets) == 0 {
+		panic("targets must not be empty")
+	}
+
+	var (
+		meta *WriteMeta
+		err  error
+	)
+	for _, target := range targets {
+		var status int
+		meta, status, err = a.updateTokenOnce(target, token, q)
+		if err == nil || status != http.StatusNotFound {
+			break
```

Comment:
I think `err` is always nil for status errors, so wouldn't this be:
```suggestion
		if err == nil && status != http.StatusNotFound {
			return meta, err
```

minor: I always prefer to `return` over `break` if possible, because it avoids needing to look at what might happen afterward. Seems like return would work here.

Reply:
Oh, actually we do check this is `requireOK`, so simply checking `err` should be fine!
```suggestion
		if err == nil || status != http.StatusNotFound {
			return meta, err
```

Reply:
Oh, nice catch! I'll go with the `&&` variation, as that's what the original code did.